### PR TITLE
appcheck

### PR DESCRIPTION
**Pull request by technoplato**
replaces https://github.com/ephemeraHQ/converse-app/pull/1369

some android physical device qol improvements: https://www.loom.com/share/867a810a8df34a29bb45845d50e01a37

## Summary by CodeRabbit

- **New Features**
	- Introduced conditional logic in the Podfile to enhance build settings for specific targets.
	- Added new dependencies for Firebase integration and build properties management.
	- Implemented a polling mechanism for join group requests to improve user experience.
	- Added a new property for appCheckDebugToken in environment configurations.
	- Introduced utility functions for API URI management and Firebase App Check integration.

- **Bug Fixes**
	- Removed the synchronization group for Google Services, streamlining project configuration.

- **Chores**
	- Restructured project groups for better organization related to Google Services.
	- Updated cocoapods version in build configurations.
	- Updated EXPO_ENV entries in development and preview configurations.
	- Enhanced error handling and security in API interactions by requiring appCheckToken.
	- Improved clarity in spam score computation logic.
	- Specified minimum required version of the Bundler gem for compatibility.

### CODE REVIEW

File: utils/mmkv.ts [hunk 1]
```diff
@@ -11,6 +11,29 @@ const storage = new MMKV();
 
 export default storage;
 
+const authMMKV = new MMKV({ id: "converse-auth" });
+
+type AuthStorageKey = "CONVERSE_REFRESH_TOKEN" | "CONVERSE_ACCESS_TOKEN";
+
+export const authMMKVStorage = {
```
Comment (alexrisch) [inline utils/mmkv.ts:18]: This should be removed

File: screens/Navigation/Navigation.tsx [hunk 2]
```diff
@@ -191,6 +194,39 @@ export function SignedInNavigation() {
   );
 }
 
+// todo: remove once integration complete
+const TestAppCheckScreen = () => {
```
[the remark this answers is not in the document]
Comment (technoplato) [reply screens/Navigation/Navigation.tsx:198]: was going to remove once Android integration done - will remove now

File: utils/mmkv.ts (hunk 1 above)
Comment (technoplato) [reply utils/mmkv.ts:18]: nice catch - leftovers from git problems

File: package.json [hunk 3]
```diff
@@ -238,6 +240,7 @@
     "eslint-plugin-react-hooks": "^5.0.0",
     "eslint-plugin-react-native": "^4.1.0",
     "eslint-plugin-unused-imports": "^4.1.4",
+    "expo-build-properties": "~0.12.5",
```
[the remark this answers is not in the document]
Comment (technoplato) [reply package.json:243]: does this dep go in devdep or regs

File: utils/getEnv.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * getEnv()
+ * /// Output:
+ * // - 'dev' when in development
+ * // - 'preview' when in preview
+ * // - 'prod' when in production
+ */
+export const getEnv = (): Environment => {
+  // todo(lustig): type .env variables
+  // @ts-ignore
+  const isExpoEnvDev = process.env.EXPO_ENV === "dev";
```
[the remark this answers is not in the document]
Comment (technoplato) [reply utils/getEnv.ts:27]: no I'm going to wait and validate at runtime with zod in a different pr

File: package.json [hunk 5]
```diff
@@ -13,7 +13,7 @@
     "ios": "expo run:ios",
     "run:ios": "EXPO_ENV=dev expo run:ios",
     "postinstall": "patch-package && node scripts/wasm.js && husky install && cross-os postinstall",
-    "start": "EXPO_ENV=dev expo start",
+    "start": "EXPO_ENV=dev expo start --scheme converse-dev",
```
Comment (technoplato) [inline package.json:16]: physical device runs weren't working without this

File: utils/apiConfig.ts
Comment (technoplato) [inline utils/apiConfig.ts:1]: this file fixes connecting to `converse-backend` on LAN via physical device

File: utils/getEnv.ts
Comment (technoplato) [inline utils/getEnv.ts:1]: i'll do better type validation on env later, just wanted to get this cleaned up for now

File: package.json [hunk 6]
```diff
@@ -5,6 +5,7 @@
     "android": "EXPO_ENV=dev expo run:android",
     "android:build:dev": "eas build -p android --profile development --local",
     "android:reverse": "adb reverse tcp:8081 tcp:8081 && adb reverse tcp:9875 tcp:9875",
+    "android:sendApk": "./scripts/android/sendApk.sh",
```
Comment (technoplato) [inline package.json:8]: some android physical device qol improvements: https://www.loom.com/share/867a810a8df34a29bb45845d50e01a37

**Review by alexrisch — APPROVED** [review]

**Pull request merged**